library/core/src/slice/mod.rs

```diff
@@ -2708,8 +2708,10 @@ impl<T> [T] {
     /// This reordering has the additional property that any value at position `i < index` will be
     /// less than or equal to any value at a position `j > index`. Additionally, this reordering is
     /// unstable (i.e. any number of equal elements may end up at position `index`), in-place
-    /// (i.e. does not allocate), and *O*(*n*) worst-case. This function is also/ known as "kth
-    /// element" in other libraries. It returns a triplet of the following from the reordered slice:
+    /// (i.e. does not allocate), and *O*(*n*) on average. In some cases this function performs *O*(*n* log *n*).
```

Comment:
"In some cases" is not very clear. I think we should explicitly say something like:

> This function performs *O*(*n*) operations on average, with a worst case of *O*(*n* log *n*) operations.